dvc/fs/fsspec_wrapper.py

```diff
@@ -37,10 +39,20 @@ def _entry_hook(self, entry):
         return entry
 
     def isdir(self, path_info):
-        return self.fs.isdir(self._with_bucket(path_info))
+        if not self.exists(path_info):
+            return False
+
+        entry = self.info(path_info)
+        return entry["type"] == "directory" or (
+            entry["name"].endswith("/")
+            and entry["type"] == "file"
+            and entry["size"] == 0
+        )
 
     def isfile(self, path_info):
-        return self.fs.isfile(self._with_bucket(path_info))
+        if not self.exists(path_info):
+            return False
+        return not self.isdir(path_info)
```

Comment:
We need a comment (maybe with a link to an issue) to clarify why we are doing such strange things.